huaweicloud/services/rds/resource_huaweicloud_rds_sqlserver_database_privilege.go

```diff
@@ -0,0 +1,401 @@
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// updateSQLServerDatabasePrivilege: update RDS SQLServer database privilege.
+	var (
+		updateSQLServerDatabasePrivilegeProduct = "rds"
+	)
+	updateSQLServerDatabasePrivilegeClient, err := cfg.NewServiceClient(updateSQLServerDatabasePrivilegeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	oldRaws, newRaws := d.GetChange("users")
+	createUsers := newRaws.(*schema.Set).List()
+	deleteUsers := oldRaws.(*schema.Set).Difference(newRaws.(*schema.Set)).List()
+
+	if len(deleteUsers) > 0 {
+		if err = deleteSQLServerDatabasePrivilege(ctx, d, schema.TimeoutUpdate, updateSQLServerDatabasePrivilegeClient,
+			deleteUsers); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if len(createUsers) > 0 {
+		if err = createSQLServerDatabasePrivilege(ctx, d, schema.TimeoutUpdate, updateSQLServerDatabasePrivilegeClient,
+			createUsers); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceSQLServerDatabasePrivilegeRead(ctx, d, meta)
+}
+
+func resourceSQLServerDatabasePrivilegeDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteSQLServerDatabasePrivilege: delete RDS SQLServer database privilege
+	var (
+		deleteSQLServerDatabasePrivilegeProduct = "rds"
+	)
+	deleteSQLServerDatabasePrivilegeClient, err := cfg.NewServiceClient(deleteSQLServerDatabasePrivilegeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	if err = deleteSQLServerDatabasePrivilege(ctx, d, schema.TimeoutDelete, deleteSQLServerDatabasePrivilegeClient,
+		d.Get("users").(*schema.Set).List()); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func createSQLServerDatabasePrivilege(ctx context.Context, d *schema.ResourceData, timeout string,
+	client *golangsdk.ServiceClient, rawUsers interface{}) error {
+	// createSQLServerDatabasePrivilege: create RDS SQLServer database privilege.
+	createSQLServerDatabasePrivilegeHttpUrl := "v3/{project_id}/instances/{instance_id}/db_privilege"
+
+	instanceId := d.Get("instance_id").(string)
+	dbName := d.Get("db_name").(string)
+
+	createSQLServerDatabasePrivilegePath := client.Endpoint + createSQLServerDatabasePrivilegeHttpUrl
+	createSQLServerDatabasePrivilegePath = strings.ReplaceAll(createSQLServerDatabasePrivilegePath, "{project_id}",
+		client.ProjectID)
+	createSQLServerDatabasePrivilegePath = strings.ReplaceAll(createSQLServerDatabasePrivilegePath, "{instance_id}",
+		d.Get("instance_id").(string))
+
+	createSQLServerDatabasePrivilegeOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	users := buildCreateSQLServerDatabasePrivilegeRequestBodyCreateUser(rawUsers)
+	start := 0
+	end := int(math.Min(50, float64(len(users))))
```

Comment:
suggest to define a const variable for the maximum of single API request.

Reply:
done